homeassistant/components/stream/__init__.py

```diff
@@ -292,8 +304,7 @@ def _run_worker(self) -> None:
         wait_timeout = 0
         while not self._thread_quit.wait(timeout=wait_timeout):
             start_time = time.time()
-
-            self._available = True
+            self.hass.loop.call_soon_threadsafe(self._async_update_state, True)
```

Comment:
Let's use `self.hass.add_job`
```suggestion
            self.hass.add_job(self._async_update_state, True)
```

It will call the same method under the hood but at least it passes through the core in case we ever need to figure out what is happening where.